src/MemoryMgmt.chpl

```diff
@@ -0,0 +1,140 @@
+module MemoryMgmt {
+
+    use Subprocess;
+    use Memory.Diagnostics;
+    use Logging;
+    
+    private config const logLevel = LogLevel.DEBUG;
+    private config const logChannel = LogChannel.CONSOLE;
+    const mmLogger = new Logger(logLevel,logChannel);
+
+    /*
+     * Indicates whether static locale host memory or dynamically-captured
+     * memory allocated to the Arkouda process is used to estimate whether
+     * sufficient memory is available to execute the requested command.
+     */
+    enum MemMgmtType {STATIC,DYNAMIC}
+
+    /*
+     * The percentage of currently available memory on each locale host 
+     * that is the limit for memory allocated to each Arkouda locale.
+     */
+    config const availableMemoryPct: real = 90;
+
+    /*
+     * Config param that indicates whether the static memory mgmt logic in
+     * ServerConfig or dynamic memory mgmt in this module will be used. The
+     * default is static memory mgmt.
+     */
+    config const memMgmtType = MemMgmtType.STATIC;
+
+    proc getArkoudaPid() : string throws {
+        var pid = spawn(["pgrep","arkouda"], stdout=pipeStyle.pipe);
```

Comment:
I don't think it matters too much (I test both and either works), but should we use `arkouda_server` instead of just `arkouda` for the process name?

Reply:
@Ethan-DeBandi99 ah, I will check that out

Reply:
@Ethan-DeBandi99 that works, will update accordingly, good idea

![image](https://github.com/Bears-R-Us/arkouda/assets/10785153/72e3f46a-c0a9-404b-9bf8-76b8a4ceb4d5)

Reply:
@pierce314159 gotcha, will add issue

Reply:
@Ethan-DeBandi99 just pushed update w/ pgrep looking for arkouda_server